Server/Repo/StudentRepo/StudentRepo.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Threading.Tasks;
+using Test.Server.Data;
+using Test.Shared;
+
+namespace Test.Server.Repo.StudentRepo
+{
+    public class StudentRepo
+    {
+        private readonly AppDbContext _context;
+
+        public StudentRepo(AppDbContext context)
+        {
+            _context = context;
+        }
+        public async Task<ActionResult<IEnumerable<Student>>> GetFullStudents(int? id)
+        {
+            var result = await _context.Students.Select(student => new Student()
```

Comment:
I would prefer using LINQ. However, maybe this query is indeed so complex that writing it in procedural way might pay off.